src/Avalonia.Controls/Control.cs

```diff
@@ -215,18 +291,117 @@ void IVisualBrushInitialize.EnsureInitialized()
         /// <returns>The control that receives the focus adorner.</returns>
         protected virtual IControl? GetTemplateFocusTarget() => this;
 
+        /// <summary>
+        /// Schedules <see cref="OnLoadedCore"/> to be called for this control.
+        /// For performance, it will be queued with other controls.
+        /// </summary>
+        internal void ScheduleOnLoadedCore()
+        {
+            // Note: This code is not multi-threaded ready and assumes to run only on the UIThread
+            if (_isLoaded == false)
+            {
+                bool isAdded = _loadedQueue.Add(this);
+
+                if (isAdded)
+                {
+                    _isLoadedProcessing = true;
```

Comment:
I had to remove the _isLoadedProcessing check here to ensure Loaded will actually be called in all cases. This completely breaks queueing functionality. So I'm about back to the beginning.

The Dispatcher code needs to check if the _loadedQueue is not empty and then re-call itself when it completes. None of this feels right so if anyone has a better idea let me know.

Reply:
The check was added back along with an ugly attempt to work around the issue.